src/http_peer.rs

```diff
@@ -171,14 +170,14 @@ impl<R:AsyncRead> WaitForHttpHead<R> {
     }
 }
 
-impl<R:AsyncRead> Future for WaitForHttpHead<R> {
+impl<R: AsyncRead> Future for WaitForHttpHead<R> {
     type Item = (WaitForHttpHeadResult, R);
     type Error = Box<dyn std::error::Error>;
 
     fn poll(&mut self) -> ::futures::Poll<Self::Item, Self::Error> {
         loop {
             if self.buf.is_none() || self.io.is_none() {
-                Err("WaitForHttpHeader future polled after completion")?;
+                return Err("WaitForHttpHeader future polled after completion".into());
```

Comment:
Why longer `return ...into()` instead of shorter `Err()?`? Is there some agreement about this?

Or are you just changing everything Clippy complains about assuming what's in Clippy is official?

Reply:
Maybe better `#![allow(clippy::try_err)]`?